web-console/future-openshift/health-checks/health-checks.md

```diff
@@ -0,0 +1,21 @@
+# Health Checks
+
+Health checks allow users to monitor various container states. The following design adds the ability to create and edit health checks via forms.
+
+## Design
+* To access health checks, users can drill down into a container from the containers list on any resource overview tab.
+![overview](img/C1.png)
+
+* On the container details page, there will be an editable Health Checks section. There are 3 types of probes users can add; each probe will be listed in the section so that users can quickly see any probes they have created or see which ones they still need to create.
+![container-overview](img/C2.png)
+
+* The health check form allows users to add a single probe of each type. Users can add any combination of probes they wish. Probes can be removed by clicking "Remove [Probe Type]".
+![form](img/C3.png)
+
+* The form fields for each probe type are the same.
+* Note that probe forms will be read-only for some resources, similar to environments. In these cases, an inline alert will redirect users to the parent resource.
+![form-long](img/C4.png)
+
+* Failing health checks will show up in the events log. They will also cause an inline alert to show on the overview tab of the related resource and in the side panel view on the Workloads tab.
+![overview](img/overview.png)
```

Comment:
Any idea if this will tie into a prometheus alert? Or maybe we should be checking for these and reporting them up to the Status card on the dashboard in some way. What do you think @andybraren ?

Reply:
@cshinn I don't think this has an alerting rule, but should it? If it does this would bubble up to dashboards, but if not we could bubble this up as a GUI-only Status Message. @bmignano how severe is a container not passing one of these health checks? It gets killed and restarted, right? Would this warning message stick around for a while, and is therefore worth surfacing even higher?

Reply:
Based on my understanding, for liveness and startup probes kubelet will kill the container and follow its restart policy to recreate it. I imagine if it keeps failing though, the alert would continue to pop up. Readiness probes seem like they would continue to fail until action is taken. These failures would show up in events as well, so presumably would surface on the dashboard in that way. I think @alimobrem had some ideas around a health check card of some sort though. Open to hearing others' thoughts!